simd/unit_tests/TestSIMD.cpp

```diff
@@ -434,9 +459,21 @@ KOKKOS_INLINE_FUNCTION void device_check_condition() {
   checker.truth(all_of(a == decltype(a)(16)));
 }
 
+template <typename Abi, typename... DataTypes>
+inline void host_check_math_ops_all_types(
+    Kokkos::Experimental::Impl::data_types<DataTypes...>) {
+  (host_check_math_ops<Abi, DataTypes>(), ...);
+}
+
+template <typename Abi>
+inline void host_check_math_ops_all_types() {
+  host_check_math_ops_all_types<Abi>(
+      Kokkos::Experimental::Impl::data_type_set());
+}
```

Comment:
I'm not sure if that would work as is, since `data_types` is simply an empty class. It wouldn't know about specified template parameters without `data_type_set`. If the goal is to reduce this into calling only one function, I think it can be done in some other way. (With aggregate initialize, I can think of doing it with ctad, but I believe ctad for aggregates is c++20.)